Automattic/wp-calypso
# Do it for me purchase flow skeleton

#### Changes proposed in this Pull Request

* The initial purchase flow implementation for the DIFM purchase flow
* Flow includes
     *  Landing page `/marketing/do-it-for-me/landing/${siteSlug}`
     *  Site Info page `/marketing/do-it-for-me/site-info/${siteSlug}`
* On submit of the site info page the premium plan and the difm lite product is added to the basket 
* Strings have not been queued for translation because the design/content is yet to be finalised and the initial release will mostly be to a small pilot of users.
* The feature is not implemented behind a feature flag because we will be piloting to only a few number of users.


Screenshots:
|      |   |
| ----------- | ----------- |
| ![image](https://user-images.githubusercontent.com/3422709/131369065-f1c323fd-b786-4ae1-86f1-5d503b2aebd3.png) |  ![image](https://user-images.githubusercontent.com/3422709/131369012-08ae670c-fd5a-4ae6-a3a5-011224a70d68.png)       |
| ![image](https://user-images.githubusercontent.com/3422709/131368990-0e0aa078-e2ca-476b-9c05-84d6b313e8f1.png)         |        |

#### Testing instructions

- Go to`/marketing/do-it-for-me/landing/${siteSlug}` 
- Click through the call to action buttons
- Make sure both the difm and premium plan are available at checkout.
